scripts/xtensa-build-zephyr.py

```diff
@@ -302,22 +278,60 @@ def check_west_installation():
 			"https://docs.zephyrproject.org/latest/getting_started/index.html")
 	print(f"Found west: {west_path}")
 
+def west_reinitialize(west_root_dir: pathlib.Path, west_manifest_path: pathlib.Path):
+	"""[summary] Performs west reinitialization to SOF manifest file asking user for permission.
+	Prints error message if script is running in non-interactive mode.
+
+	:param west_root_dir: directory where is initialized.
+	:type west_root_dir: pathlib.Path
+	:param west_manifest_path: manifest file to which west is initialized.
+	:type west_manifest_path: pathlib.Path
+	:raises RuntimeError: Raised when west is initialized to wrong manifest file
+	(not SOFs manifest) and script is running in non-interactive mode.
+	"""
+	global west_top
+	message = "West is initialized to manifest other than SOFs!\n"
+	message +=  f"Initialized to manifest: {west_manifest_path}." + "\n"
+	west_workspace_directory = pathlib.Path(west_root_dir.resolve(), ".west")
+	if args.no_interactive:
+		message += f"Try deleting {west_workspace_directory} directory and rerun this script."
+		raise RuntimeError(message)
+	question = message + "Reinitialize west to SOF manifest? [Y/n] "
```

Comment:
This is really overkill. Just print the two `rm-rf .west` `west init` command sthat people must run in this case and done. These two commands alo gives them a chance to understand what's going on if they're interested.

Also don't forget this script is heavily used in non-interactive CI

Reply:
@marc-hb I'm sorry but we are starting to run circles with this code review.
You suggested me that a path containing .west directory is a implementation detail and should not be used.
I tried not to use it but turns out that there is no other way to deinitialize west when set to invalid manifest X.
So I've made this user prompt to hide this implementation detail. To have it working with CI I also added no-interactive flag. Now you:
* write me that the work I've already done is an overkill
* tell me to just print `rm -rf .west` which was implementation detail before and had to be hidden/not used.
* Write:

> Also don't forget this script is heavily used in non-interactive CI

I mean, are you reading my patch notes? I wrote there:

> to avoid user prompt in CI and programmatical calls of the script added new flag --no-interactive

@marc-hb please stop:

- brainstorming on review before you read content
- brainstorming before you test, check how it works, like really think it over

I do take my time to think through solutions I propose here. Please also do so. Take you time before you write a review and when you do please make it quality information. We are getting nowhere with throwing shallow remarks 'don't do this or that' and we need this changeset to be merged in some reasonable time line as it is needed for incoming Intel ADSP Meteorlake platform.

Reply:
Let's stop for a while and think this trough.
We can always put my work to garbage saying that it is overkill, but since the job had been already done I would reconsider. To me it does look like a useful thing - you answer 'Y' to prompt and do not have to stop script execution remove .west folder and start again.
The only downfall I see now is that user does not know that .west directory is being deleted and west tool does not have anything related to "reinitialization" in its docs (and running `west init -l <other_dir>` on a already initailized manifest results in error not in reinitialization).
I would proceed with leaving existing user prompt and appending to it additional line of message:
"this will remove {some_path/.west} directory"
Do you agree or we drop the idea of user prompts?

Reply:
I initially did not realize that you were concerned about "de-initializing" the west workspace, something which seems unfortunately missing from the west documentation :-( In that case then yes I'm afraid that deleting the `.west/` "implementation detail" seems to be the only, non-documented way. Still, because it's undocumented then better just ask people to do it manually instead of the script deleting it. Scripting `rm -rf $path` is never a good idea, imagine when a bug creeps in how `$path` is made!